app/js/components/homePage/manageApps/manageApps.jsx

```diff
@@ -54,22 +58,80 @@ export default class ManageApps extends React.Component {
     const url = location.href.split('/')[3];
     let apiBaseUrl = `/${applicationDistribution}/${url}/ws/rest`;
     this.requestUrl ='/owa/addapp';
+    let that = this;
```

Comment:
Why did you do this?

Reply:
I created an alias to the this keyword, binding the 'handleProgress' function to the scope (local) of the 'xhr' function where I access the ajaxSettings objects and get the progress event listener. The 'progress' event is then passed on to the handleProgress function where percentage upload is calculated.

Reply:
nice one!

Reply:
Just to add a bit more context `this.setState` is a function in the global scope. The reason why `that` was used is because the `xhr` function is creating its own scope and refused to allow access to the global scope so we cannot use methods contained in the `global` scope. The work around was to create a `reference` to the global scope i.e `that`(we could have used any other word) and call it whenever we wanted to get access to the global scope from inside any function/method that created its own scope. 
@paulupendo I hope I'm correct.

Reply:
@andela-jomadoye Very correct 🐒

Reply:
@paulupendo, @andela-jomadoye Thank you for the explanation, I now understand